parser_test.go

```diff
@@ -6244,3 +6244,55 @@ func (s *testParserSuite) TestPlanRecreator(c *C) {
 	c.Assert(v.Stmt.Text(), Equals, "SELECT a FROM t")
 	c.Assert(v.Analyze, IsTrue)
 }
+
+func (s *testParserSuite) TestGBKEncoding(c *C) {
+	p := parser.New()
+	gbkEncoding, _ := charset.Lookup("gbk")
+	encoder, decoder := gbkEncoding.NewEncoder(), gbkEncoding.NewDecoder()
+	sql, err := encoder.String("create table 测试表 (测试列 varchar(255) default 'GBK测试用例');")
+	c.Assert(err, IsNil)
+
+	stmt, err := p.ParseOneStmt(sql, "", "")
+	c.Assert(err, IsNil)
+	checker := &gbkEncodingChecker{}
+	_, _ = stmt.Accept(checker)
+	c.Assert(checker.tblName, Not(Equals), "测试表")
+	c.Assert(checker.colName, Not(Equals), "测试列")
+	c.Assert(checker.expr, Not(Equals), "GBK测试用例")
+
+	p.SetParserConfig(parser.ParserConfig{Decoder: decoder})
+	stmt, err = p.ParseOneStmt(sql, "", "")
+	c.Assert(err, IsNil)
+	_, _ = stmt.Accept(checker)
+	c.Assert(checker.tblName, Equals, "测试表")
+	c.Assert(checker.colName, Equals, "测试列")
+	c.Assert(checker.expr, Equals, "GBK测试用例")
+}
+
+type gbkEncodingChecker struct {
```

Comment:
what is this used for, did not see any usage in this PR?

Reply:
This is used in the test.